File: allfolio-backend/unified-asset/src/main/kotlin/com/allfolio/unifiedasset/domain/goal/Goal.kt
package com.allfolio.unifiedasset.domain.goal

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.UUID

enum class GoalCategory {
    RETIREMENT, HOUSING, EDUCATION, TRAVEL, EMERGENCY, OTHER
}

data class Goal(
    val id: UUID,
    val userId: UUID,
    val name: String,
    val description: String?,
    val targetAmount: BigDecimal,
    val targetDate: LocalDate?,
    val category: GoalCategory,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
) {
    companion object {
        fun create(
            userId: UUID,
            name: String,
            description: String?,
            targetAmount: BigDecimal,
            targetDate: LocalDate?,
            category: GoalCategory,
        ) = Goal(
            id = UUID.randomUUID(),
            userId = userId,
            name = name,
            description = description,
            targetAmount = targetAmount,
            targetDate = targetDate,
            category = category,
            createdAt = LocalDateTime.now(),
            updatedAt = LocalDateTime.now(),
        )
    }
}

File: frontend/allfolio_app/types/goal.ts
export type GoalCategory = 'RETIREMENT' | 'HOUSING' | 'EDUCATION' | 'TRAVEL' | 'EMERGENCY' | 'OTHER'

export interface GoalRequest {
  name: string
  description?: string
  targetAmount: number
  targetDate?: string | null
  category: GoalCategory
}

export interface GoalResponse {
  id: string
  userId: string
  name: string
  description?: string
  targetAmount: number
  targetDate?: string
  category: GoalCategory
  currentAmount: number
  progressPct: number
  remainingAmount: number
  daysRemaining?: number
  createdAt: string
  updatedAt: string
}

export interface GoalsResponse {
  goals: GoalResponse[]
  totalNav: number
  generatedAt: string
}
